### ISSUE TITLE
Filter on nested properties

### ISSUE THREAD

**Issue by efikarra** [djett41/ionic-filter-bar#31]
Hello, 

i have an array of objects and each one has some nested objects. I would like to filter on nested properties but filtering using filterExpression seems to filter only on first level properties. I would like to do something like that : filterExpression : "nestedObject.property" but it does not work!
Is that possible;

Thanks!

**Comment by djett41** [comment]
@efikarra Make sure you are passing your expression in an `expression` propert rather than `filterExpression`.  If that's not your issue, take a look at the filter [docs](https://docs.angularjs.org/api/ng/filter/filter) which goes over nested objects in expressions and double check that you are passing in the correct expression. 

 I'm just delegating to angulars filter logic when expression is passed in so anything that works out of the box for angular will work here.

**Comment by efikarra** [comment]
My issue with expression property is that every expression that i have seen so far in the docs, seems to require the filterText. For example an expression for my case could be {nestedObject: {property: filterText}} which i checked that works as expected when i used a fixed string instead of filterText.
 However, i cannot find how to access the filterText in expression property!

**Comment by djett41** [comment]
Upgrade to the new version 1.0.6, filterText has been added to the expression function as the first argument.  See the unit test as an example!  Thanks!

**Comment by efikarra** [comment]
It works perfectly, thanks a lot!

**Comment by EricDosReis** [comment]
@efikarra can you show your code? I'm have the same issue, but I don't understand how implement this solution.

**Comment by efikarra** [comment]
let's say that you have the list:  $scope.items= [{subItem: {description: } }, {subItem: {description: } },...] and you want to filter on nested property description which is a string. Then the code is:

```
$scope.showFilterBar = function () {
   var filterBarInstance = $ionicFilterBar.show({
     items: $scope.items,
     update: function (filteredItems) {
         $scope.items= filteredItems;
     },
     expression: function (filterText, value, index, array) {
         return value.subItem.description.indexOf(filterText) !== -1
     }
   });
 };
```

**Comment by EricDosReis** [comment]
Thanks @efikarra for answer my question, but in my case I have a list of categories objects with a list of products objects inside. And we already found a solution, look the code:

``` json
[{
    "Arm. Deslizantes": [{
        "FK_ITEM": "12",
        "NAME_ITEM": "Assistencia Tecnica",
        "ACTIVE": "1",
        "PRICE": "0"
    }, {
        "FK_ITEM": "7",
        "NAME_ITEM": "Estanteria",
        "ACTIVE": "1",
        "PRICE": "0"
    }, {
        "FK_ITEM": "14",
        "NAME_ITEM": "Expression",
        "ACTIVE": "1",
        "PRICE": "0"
    }, {
        "FK_ITEM": "18",
        "NAME_ITEM": "dasdsdasd",
        "ACTIVE": "1",
        "PRICE": "2.22"
    }],
    "Serviços": [{
        "FK_ITEM": "19",
        "NAME_ITEM": "TESTE LALALA",
        "ACTIVE": "1",
        "PRICE": "10"
    }, {
        "FK_ITEM": "16",
        "NAME_ITEM": "teste",
        "ACTIVE": "1",
        "PRICE": "2"
    }]
}]
```

``` javascript
.service('filterService', function($ionicFilterBar, $filter) {
    return function(scope, listName, keyNestedList) {
        $ionicFilterBar.show({
            items: scope[listName],
            cancelText: 'Cancelar',
            update: function(filteredItems, filterText) {
                if (keyNestedList)
                    for (var key in filteredItems) {
                        var value = filteredItems[key]
                          , filtered = $filter('filter')(value[keyNestedList], filterText)
                          , aux = {};

                        for (var k in value)
                            aux[k] = value[k];

                        if (filtered != '')
                            aux[keyNestedList] = filtered;

                        filteredItems[key] = aux;
                    }

                scope[listName] = filteredItems;
            }
        });
    }
})
```

**Comment by djett41** [comment]
@EricDosReis if you add an expression, the expression will be invoked for every item in the items you send.  Then when you return true from the expression, the item gets added to the list of filtered items.. 

You will need to do one of the following, either add an expression for your filter logic, or add a custom filter and pass that in.

**Comment by jrub** [comment]
Hi there, as an angular newbie I have found somewhat difficult to make search work with nested properties (solved it thanks to this issue). I think it would be great to have `filterProperties` work with nested properties, because it seems a natural way to use it.

If this is not possible right away, would be nice to have this at least mentioned on the `filterProperties` usage paragraph into the readme.

Also, note that `filterProperties` worked case insensitive for me, but using `expression`, I had to use `toLowerCase()`, like:

```
      expression: function (filterText, value, index, array) {
         return value.myProperty.myNestedProperty.toLowerCase().indexOf(filterText.toLowerCase()) !== -1
      }
```
